core/error.go

```diff
@@ -177,3 +177,17 @@ func IsValueTransferErr(e error) bool {
 	_, ok := e.(*ValueTransferError)
 	return ok
 }
+
+type GasLimitErr struct {
+	Have, Diff *big.Int
```

Comment:
Diff implies it's difference between two values, whereas in SubGas it's set to the value requested. How about "requested" or some such?